### PR TITLE
Add Algolia search

### PR DESCRIPTION
**Pull request by pixelastic**
As [rapidly discussed](https://twitter.com/pixelastic/status/640938034041593856) on Twitter, here is my proposal for adding search to the babeljs website.

Screencast

You can [test it live here](https://babeljs.algolia.com/).

This uses our [jekyll plugin](https://github.com/algolia/algoliasearch-jekyll) for pushing data to your [Algolia](https://www.algolia.com/) index, and our [autocomplete.js](https://github.com/algolia/autocomplete.js) library (inspired by typeahead) for displaying the results. All the ranking configuration is available in the `_config.yml` (which mirrors options available in our [REST API](https://www.algolia.com/doc/rest)), and in the `search.rb` file.

We will of course give you full access to your account, and you'll be available to edit all the settings as you see fit (just send me a PM with the email address you would like associated with) . We just ask that you keep the Algolia logo in the results.

Tell me what you think.

### CODE REVIEW

**Comment by pixelastic** [comment]
Any updates? Is that something you'd be interested in?

**Comment by thejameskyle** [comment]
Would you mind collapsing the search all the time? The header is really cramped otherwise

**Comment by pixelastic** [comment]
Here you go: https://babeljs.algolia.com/

image

image

File: Gemfile [hunk 1]
```diff
@@ -1,4 +1,8 @@
 source 'https://rubygems.org'
 
-gem 'jekyll'
+gem 'jekyll', '~> 2.5.3'
 gem 'jekyll-redirect-from'
+
+group :jekyll_plugins do
+  gem 'algoliasearch-jekyll', '~> 0.4.3'
```
Comment (thejameskyle) [inline Gemfile]: Wait, how does this work with GitHub pages? Does this require a build step?

Comment (pixelastic) [inline Gemfile]: Our plugin is not in the list of official plugins endorsed by GitHub. You have to do a `jekyll algolia push` locally whenever you do a `git push` to `gh-pages` to keep data in your search index in sync with data really displayed in your site.

**Comment by thejameskyle** [comment]
Talked about this with @sebmck and we're going to pass on this. Thank you for putting so much effort into it.

**Comment by pixelastic** [comment]
Oh...

Would you mind sharing the reasons? See if there is anything I could do. I'd really like to see this merged, as I'm now using it everytime I need to search the babeljs docs. Whatever your reasons, I'd respect them, I just want to know if there is anything I overlooked and any way to make this better.

**Comment by sebmck** [comment]
It requires an external build step to update site content which isn't something we want to introduce. We also aren't comfortable promoting another business in this matter because as to date Babel has not taken any stance in relation to corporate sponsorship or promotion.

**Comment by pixelastic** [comment]
Ok, makes sense.

We can definitely remove the logo from the search results if you find it too intrusive. A simple note with a link somewhere in an acknowledgment/thank you page is enough. We'll still give you a free and unlimited account, no worries.

We might introduce an automatic update of data pushed to `gh-pages` in the future, something based on github webhooks, but we're not there yet. If that's something you'd be confortable with, I'll ping you if (when) it's ready.

**Comment by pixelastic** [comment]
Just a quick note to let you know that we recently got a [similar PR](https://github.com/eslint/eslint.github.io/pull/152) merged into the [eslint.org](https://eslint.org) website.

We managed to completly automate the re-indexing of data through Travis. Travis listens to pushes on `gh-pages` and will launch the `jekyll algolia push` command from there. You just have to configure an `ALGOLIA_API_KEY` environment variable in your Travis config. I've [documented the process](https://github.com/algolia/algoliasearch-jekyll#github-pages) in the readme.

**Comment by hzoo** [comment]
We should do this again! - react, eslint, yarn all have a search functionality as well and it is missing from the babel website for sure

**Comment by pixelastic** [comment]
Hello,

React, Eslint and Yarn are all using Algolia for their search. Eslint is using an approach similar to the one I suggested here (ie. through a Jekyll plugin) while React and Yarn are using our [DocSearch](https://community.algolia.com/docsearch/) crawler.

I would suggest using DocSearch for the Babel doc as well. As it is based on crawling, you won't have any new step to add to your build process, we will just automatically re-index all the content every 24 hours. 

If interested, you just need to fill in your email and website url in [this form](https://community.algolia.com/docsearch/). We will configure the crawler on our side (basically defining pages to crawl and CSS selectors to extract title and content) and get back to you with a few JS lines to bind the dropdown to an input.

Let me know if this would work for you or if you need any more advice, I'd be happy to help.

**Comment by hzoo** [comment]
Ok sounds good to me

**Comment by pixelastic** [comment]
Ah yes, true. We will update the link once it's implemented on the official website

**Comment by pixelastic** [comment]
@maxiloc just pushed #977 which adds the search in the header through DocSearch. Let us know what you think.

**Pull request closed without merging**